libs/ml/examples/mnist_estimator/main.cpp

```diff
@@ -57,9 +57,12 @@ int main(int ac, char **av)
   estimator_config.print_stats    = true;
 
   // setup dataloader
-  auto       data_loader_ptr = std::make_shared<DataLoaderType>(av[1], av[2]);
-  TensorType test_label      = (data_loader_ptr->GetNext()).first;
-  TensorType test_input      = (data_loader_ptr->GetNext()).second.at(0);
+  auto data_loader_ptr = std::make_shared<DataLoaderType>(av[1], av[2]);
+  data_loader_ptr->SetTestRatio(0.2f);
+
+  auto       test_data  = data_loader_ptr->GetNext();
```

Comment:
i think this gets a piece of training data right?
In any case, wouldn't it be better to use your new functionality instead of doing this and call Validate?